mlflow/data/delta_dataset_source.py

```diff
@@ -96,6 +140,10 @@ def _to_dict(self) -> Dict[Any, Any]:
             info["delta_table_version"] = self._delta_table_version
         if self._is_databricks_uc_table():
             info["is_databricks_uc_table"] = True
+            if self._delta_table_id:
+                info["delta_table_id"] = self._delta_table_id
+            else:
+                info["delta_table_id"] = self._lookup_table_id(self._delta_table_name)
```

Comment:
if this lookup fails due to say 500 error, do we want this dataset logging to terminate? logged datasets are immutable, so this would result in an "incorrect" record logged.

Reply:
My intuition is to not fail logging if the id lookup fails.